src/middleware.ts

```diff
@@ -248,7 +283,15 @@ export const persist = <S extends State>(
       blacklist.forEach((key) => delete state[key])
     }
 
-    return storage?.setItem(name, await serialize({ state, version }))
+    return thenableSerialize({ state, version }).then((serializedValue) => {
+      if (storage) {
```

Comment:
We are sure `storage` is defined here. Can we remove the `if`? Does TS complain?

Reply:
the `ts` complain about it. and I agree with `ts`😆 .
because it may be an async call.

Reply:
Let's either a) assert type or b) define `const` after early return.

Reply:
if we define it as `const` in `line:254` then we lose the variable scope.
type assertion seems like a good option.
```Typescript
 storage!.setItem(name, serializedValue)
```

Reply:
I meant after L270. But, it increases JS code. Let's do type assertion.
```ts
(storage as StateStorage).setItem(name, serializedValue)
```
I prefer this with no reason. Maybe some eslint rules disliked Non-null Assertion (!)